esphome/components/mcp4728/mcp4728_output.cpp

```diff
@@ -0,0 +1,121 @@
+#include "mcp4728_output.h"
+
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace mcp4728 {
+
+static const char *const TAG = "mcp4728";
+
+void MCP4728Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MCP4728 (0x%02X)...", this->address_);
+  auto err = this->write(nullptr, 0);
+  if (err != i2c::ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+}
+
+void MCP4728Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "MCP4728:");
+  LOG_I2C_DEVICE(this);
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "Communication with MCP4728 failed!");
+  }
+}
+
+void MCP4728Component::loop() {
+  if (this->update_) {
+    this->update_ = false;
+    if (this->store_in_eeprom_) {
+      if (!this->seq_write_()) {
+        this->status_set_error();
+      } else {
+        this->status_clear_error();
+      }
+    } else {
+      if (!this->multi_write_()) {
+        this->status_set_error();
+      } else {
+        this->status_clear_error();
+      }
+    }
+  }
+}
+
+void MCP4728Component::set_channel_value_(MCP4728ChannelIdx channel, uint16_t value) {
+  uint8_t cn = 0;
+  if (channel == MCP4728_CHANNEL_A) {
+    cn = 'A';
+  } else if (channel == MCP4728_CHANNEL_B) {
+    cn = 'B';
+  } else if (channel == MCP4728_CHANNEL_C) {
+    cn = 'C';
+  } else {
+    cn = 'D';
+  }
+  ESP_LOGD(TAG, "Setting MCP4728 channel %c to %d!", cn, value);
```

Comment:
```suggestion
  ESP_LOGV(TAG, "Setting MCP4728 channel %c to %d!", cn, value);
```

For example this output is attached to a light, the logs would just be spammed during a transition